src/components/[guild]/hooks/useUser.ts

```diff
@@ -90,11 +101,16 @@ const useUserPublic = (
               "You've connected your account from a new device, so you have to sign a new message to stay logged in",
             duration: 5000,
           })
-
-          setIsWalletSelectorModalOpen(true)
         }
       }
 
+      // If we didn't set the keyPair field, the user either doesn't have one locally, or has an invalid one
+      const shouldOpenWalletSelectorModal = !user.keyPair
+
+      if (shouldOpenWalletSelectorModal && !ignoredRoutes.includes(router.route)) {
```

Comment:
nit, but I think the comment explains this logic pretty well, so we don't need an extra variable here, we can just check `if(!user.keyPair)`, wdyt?

Reply:
Agreed, the comment should be enough, thanks for pointing it out!
Changed in https://github.com/guildxyz/guild.xyz/pull/1249/commits/5e1a3d66d989996957574e36a710ce330918d1a0